src/components/MessageInput/hooks/useCooldownTimer.tsx

```diff
@@ -40,13 +40,24 @@ export const useCooldownTimer = <
         Math.max(0, (new Date().getTime() - ownLatestMessageDate.getTime()) / 1000)
       : undefined;
 
-    setCooldownRemaining(
+    const remaining =
       !skipCooldown &&
-        typeof timeSinceOwnLastMessage !== 'undefined' &&
-        cooldownInterval > timeSinceOwnLastMessage
+      typeof timeSinceOwnLastMessage !== 'undefined' &&
+      cooldownInterval > timeSinceOwnLastMessage
         ? Math.round(cooldownInterval - timeSinceOwnLastMessage)
-        : 0,
-    );
+        : 0;
+
+    setCooldownRemaining(remaining);
+
+    if (!remaining) return;
+
+    const timeout = setTimeout(() => {
+      setCooldownRemaining(0);
+    }, remaining * 1000);
```

Comment:
Is this timeout here to reset the "global" cooldown remaining? And the timeout in `CooldownTimer.tsx` is there only for updating the UI?

Can we move everything here and use `cooldownRemaining` in the Component? Are we avoiding this to save on re-renders?